types/immutability-helper/index.d.ts

```diff
@@ -27,8 +28,8 @@ interface UpdateArraySpec extends UpdateSpecCommand {
 type CommandHandler = (specValue: any, originalValue: any) => any;
 
 interface UpdateFunction {
-    (value: any[], spec: UpdateArraySpec): any[];
-    (value: {}, spec: UpdateSpec): any;
+    <K extends Array<T>, T>(value: T[], spec: UpdateArraySpec): T[];
```

Comment:
`K` not used. You might add a `tslint.json` containing `{ "extends": "dtslint/dt.json" }` to catch this kind of error.

Reply:
Given that the test code above pulled from the React docs has a heterogeneous array, that's probably something we want to support? Perhaps `<K extends Array<any>>(value: K, ...): K`?

Reply:
`<K extents Array<any>>` will cause type erasure. There is no way to specify a generic as `<_ extends Array<T>`.  So either K or T will be unused unless i cast value as `K` and return as `T[]`

Reply:
looks like `<_ extends Array<T>` makes intellij happy, unfortunately the project won't build locally at all so i'll wait for another travis run.
If you pass in an `Array<any>` or a heterogeneous array you will get out the same, but it won't erase the type of a typed array.

Reply:
I don't know what you mean by "cause type erasure". Can you show me examples where the types don't behave as expected if you do what Andy or I suggest? (Also, I should have named the type variable `T` in my example, not `K`, since there is no other type variable).

Reply:
To clarify, my suggestion would be to simply remove `K`.